src/statistical/unigram.jl

```diff
@@ -0,0 +1,177 @@
+"""
+struct Sentencepiecemodel
+  vocab::Array{String,1}
+  logprob::Array{Float64,1}
+end
+structure, To hold vocabulary,log probability and index  
+"""
+struct Sentencepiecemodel
+  vocab::Array{String,1}
+  logprob::Array{Float64,1}
+end
+
+function load(ty::Type{T}, name::String) where T<:PretrainedTokenizer
+    filepath = @datadep_str name
+    filepath = "$filepath/$name"
+    load(filepath)  
+end
+    
+function load(path)
+    vocab = readlines(path)
+    vocabnew = split.(vocab, "\t")
+    vo = []
+    for i in 1:length(vocab)
+        vocab1 = vocabnew[i][1]
+        vocab1 = replace(vocab1, "▁"=>"_")
+        push!(vo, vocab1)
+    end
+    vocab1 = convert(Array{String,1},vo)
+    logprob = []
+    for i in 1:length(vocab)
+        logp = vocabnew[i][2]
+        push!(logprob, logp)    
+    end
+    logp = convert(Array{String,1}, logprob)
+    logp =  parse.(Float64, logprob)
+    spm = Sentencepiecemodel(vocab1, logp)
+    return spm
+end
+
+# to get index of given string
+function getindex(sp::Sentencepiecemodel, text)
+    id_list = findall(x->x==text, sp.vocab)
+    length(id_list) == 0 && return 2 #unk token index 
+    return id_list[1]
+end
+
+"""
+struct Nodes 
+    text::String
+    score::Float32
+    index::Int64
+    start::Int
+    en::Int
+end
+Utility structure, To hold the results of the forward pass (the forward Viterbi lattice)
+hold the token token string, score, vocabulary index, start and end character position   
+"""
+struct Nodes 
+    text::String
+    score::Float32
+    index::Int64
+    start::Int
+    en::Int
+end
+
+"""
+    decode_forward(sp::Sentencepiecemodel,text::String)
+Return all possible ngrams generated from sequence of items, as an Array{String,1}
+# Example
+```julia-repl
+julia> seq = ["To","be","or","not"]
+julia> a = everygram(seq,min_len=1, max_len=-1)
```

Comment:
Where is `everygram` defined?
The docstrings do not match function.

Reply:
changed